invenio_app_ils/records/api.py

```diff
@@ -188,3 +191,31 @@ class Keyword(IlsRecord):
     def create(cls, data, id_=None, **kwargs):
         """Create Keyword record."""
         return super(Keyword, cls).create(data, id_=id_, **kwargs)
+
+
+class Patron():
+    """Patron record class."""
+
+    _index = "patrons-patron-v1.0.0"
+    _doc_type = "patron-v1.0.0"
+
+    def __init__(self, id, revision_id=None):
+        """Create a `Patron` instance.
+
+        Patron instances are not stored in the database
+        but are indexed in elasticsearch.
+        """
+        self.user = User.query.filter_by(id=id).one_or_none()
+        self.id = self.user.id if self.user else ''
```

Comment:
Will an exception be raised if no user was found with `id`? If it does, do we need to check if there is a user for `self.id`?

Reply:
Hey you were right so I force to return one result which will cause a `NoResultFound` if it fails. So we can prevent unknown users to be indexed. WDYT?